app/services/discord/guild_registrar.rb

```diff
@@ -0,0 +1,38 @@
+module Discord
+  class GuildRegistrar
+    attr_reader :interface
+
+    def initialize(guild)
+      @guild = guild
+    end
+
+    def call
+      register_guild
+    end
+
+    private
+
+    attr_reader :guild
+
+    def register_guild
+      @interface = Interface.find_by(
+        interface_type: :discord_guild,
+        external_id: guild.id
+      )
+
+      return true if @interface.present?
+
+      create_new_interface
+    end
+
+    def create_new_interface
+      @interface = Interface.new(
+        interface_type: :discord_guild,
+        external_id: guild.id
+      )
+      raise InvalidGuild, interface unless @interface.valid?
+
+      @interface.save!
+    end
```

Comment:
This can be way more simple, look:

```suggestion
    def call
      @interface = Interface.find_or_create_by!(
        interface_type: :discord_guild,
        external_id: guild.id
      )
    rescue ActiveRecord::RecordInvalid => e
      raise InvalidGuild, e.record
    end

    private

    attr_reader :guild
```

Reply:
Done!